### ISSUE TITLE
VoiceOver on iOS

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#10185]
VoicerOver on iOS (for accessibility) is broken. This bug is here to collect all the various ways it is broken.

An incomplete list of issues:
- [x] accessibility frame is not drawn at the right position around a widget (https://github.com/flutter/engine/pull/3700)
- [x] scrolling doesn't work (https://github.com/flutter/engine/issues/3701)
- [x] activating a button to switch screens and scrolling crashes app
- [x] when the side drawer is open, elements covered by the drawer are still selectable
- [x] turning voiceover on/off while in a flutter app crashes it
- [x] can't close side drawer

Ultimate goal: make flutter_gallery fully accessible on iOS with VoiceOver.

**Comment by goderbauer** [comment]
Current status: I am able to navigate the flutter_gallery app with iOS accessibility tools without crashes. I can reach every element that is reachable and I can't reach elements that shouldn't be reachable. With that, the initial goal of this bug is archived. There is still work to do to make iOS accessibility great. That will be tracked in separate bugs.

**Comment by shusain93** [comment]
Is there another issue for actually fixing fixing a11y scrolling on iOS? While it is possible to scroll (three finger up down) the automatic scrolling of a list does not work (like turn on voiceover on mail, you can keep swiping from left to right until you reach the bottom of the list, even if it's not visible). This does not work in flutter which will only scroll like this until the next item would be off screen.

The way that I think regular UITableViews are handled is that the system will actually scroll up a little bit so that the next item (if it exists) is always in view. This means actually that flutter is most of the way there because it's stopping correctly, assuming that the list couldn't be scrolled anymore, however we just need to scroll the next one in to view automatically on accessibility navigation.

**Comment by goderbauer** [comment]
@shusain93 That problem is tracked in https://github.com/flutter/flutter/issues/11983. I am working on implementing it, but it is a bit tricky.
